File: Week_02/Practicum/Merge Sorted Array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int end1 = m - 1;
        int end2 = n - 1;
        
        int curr = m + n - 1;
        
        while (end1 >=0 && end2 >= 0) {
            if (nums1[end1] > nums2[end2] ) 
                nums1[curr--] = nums1[end1--];
            else
                nums1[curr--] = nums2[end2--];
        }

        while(end1 >=0) {
            nums1[curr--] = nums1[end1--];
        }

        while(end2 >=0) {
            nums1[curr--] = nums2[end2--];
        }
    }
};

File: week 02/Practicum/Python/Merge Sorted Array.py
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        n1_idx = m - 1
        n2_idx = n - 1
        r_idx = m + n - 1

        while n2_idx >= 0:
            if n1_idx >= 0 and nums1[n1_idx] > nums2[n2_idx]:
                nums1[r_idx] = nums1[n1_idx]
                n1_idx -= 1
            else:
                nums1[r_idx] = nums2[n2_idx]
                n2_idx -= 1

            r_idx -= 1